src/consensus/vote.cpp

```diff
@@ -288,23 +290,44 @@ std::vector<Vote> VoteManager::getVerifiedVotes(uint64_t const pbft_round, size_
   std::vector<Vote> verified_votes;
   auto votes_to_verify = getUnverifiedVotes();
 
+  h256 latest_final_chain_block_hash = final_chain_->get_last_block()->hash();
+
+  if (latest_final_chain_block_hash != current_period_final_chain_block_hash_) {
+    current_period_final_chain_block_hash_ = latest_final_chain_block_hash;
+    if (!votes_invalid_in_current_final_chain_period_.empty()) {
+      LOG(log_dg_) << "After new final chain block, will now re-attempt to validate "
+                   << votes_invalid_in_current_final_chain_period_.size() << " unverified PBFT votes";
+      votes_invalid_in_current_final_chain_period_.clear();
+    }
+  }
+
   for (auto const& v : votes_to_verify) {
+    if (std::find(votes_invalid_in_current_final_chain_period_.begin(),
+                  votes_invalid_in_current_final_chain_period_.end(),
+                  v.getHash()) != votes_invalid_in_current_final_chain_period_.end())
```

Comment:
Could use std::unordered_set for votes_invalid_in_current_final_chain_period_. 
votes_invalid_in_current_final_chain_period_.count(v.getHash()) is in constant time